modules/cactus-web/src/Footer/Footer.tsx

```diff
@@ -0,0 +1,208 @@
+import PropTypes from 'prop-types'
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import styled, { DefaultTheme, StyledComponent } from 'styled-components'
+
+import { boxShadow } from '../helpers/theme'
+import Link from '../Link/Link'
+import { ScreenSizeContext, Size } from '../ScreenSizeProvider/ScreenSizeProvider'
+
+interface FooterContextType {
+  addLink: (content: React.ReactNode, to: string) => void
+}
+
+interface LinkType {
+  content: React.ReactNode
+  to: string
+}
+
+interface FooterState {
+  links: LinkType[]
+}
+
+interface LinkColProps {
+  maxCols: number
+}
+
+interface FooterProps extends React.HTMLAttributes<HTMLDivElement> {
+  logo?: string | React.ComponentType<any>
```

Comment:
Since you can't pass props, you should probably make this a `React.ReactElement<any>` instead, that way they have more flexibility in what kinds of components they can use.

Reply:
Did you mean something like `(props: any) => React.ReactElement`? TS doesn't seem to like me rendering the logo component when its type is set to `React.ReactElement<any>`. I keep getting the error: `JSX element type 'Logo' does not have any construct or call signatures`.

Reply:
No, I mean they should pass the instantiated component instead of the type, i.e.
```
<Footer logo={<MyAwesomeLogo aspectRatio="wide" />}>
```
so the type here would be
```suggestion
  logo?: string | React.ReactElement
```

Reply:
Ah, that's what I was missing. Gotcha.